### PR TITLE
add option to force deduplicate children

### PR DESCRIPTION
**Pull request by urykhy**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

reduce dataloss chance on insert into table with mv.

Detailed description / Documentation draft:

if you insert data into replicated table with mv, there is a chance, that ZooKeeper exception can cause data inserted into table only, since if CH do not try to insert into mv if block already deduplicated by main table. Usually this works well, but if ZooKeeper error occures after insert into main table, this block will never reach MV. If u set `force_deduplicate_childrens=1`, then CH will try to insert block into MV even if it deduplicated by parent table.

partial fix for #2621

### CODE REVIEW

**Comment by akuzm** [comment]
Is it possible to add some kind of integration test for this?

**Comment by urykhy** [comment]
@akuzm i'm not sure

we arrange test environment to trigger this kind of error, but:
* it's build from closed components (uploader, tests, test data)
* not 100% accuracy, this patch only reduce rate of error. it's still possible that data will not reach MV or reach MV twice.
* time to run up to 60 minutes.

**Comment by den-crane** [comment]
@urykhy at least you can emulate MV insert fail
https://gist.github.com/den-crane/bc35a15a8d71899fe15e5ed36668d387

File: dbms/src/Core/Settings.h [hunk 1]
```diff
@@ -395,6 +395,7 @@ struct Settings : public SettingsCollection<Settings>
     M(SettingBool, enable_early_constant_folding, true, "Enable query optimization where we analyze function and subqueries results and rewrite query if there're constants there", 0) \
     \
     M(SettingBool, partial_revokes, false, "Makes it possible to revoke privileges partially.", 0) \
+    M(SettingBool, force_deduplicate_childrens, false, "If the 'root' table deduplicates blocks, there are no need to make deduplication for children. Use true to always deduplicate childrens.", 0) \
```
Comment (alexey-milovidov) [inline dbms/src/Core/Settings.h]: childrens -> children

Comment (alexey-milovidov) [reply dbms/src/Core/Settings.h]: Need explanation that children are denendent materialized views.

force_deduplicate_childrens -> `deduplicate_blocks_in_dependent_materialized_views`

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
01083_expressions_in_engine_arguments
- well known issue.

**Pull request merged**